testing/test_sct_segment_graymatter.py

```diff
@@ -69,34 +56,44 @@ def test_integrity(param_test):
     null_slices = []
     gm_dice = []
     for line in gm_dice_lines:
-        n_slice, dc = line.split(' ')
-        # remove \n from dice result
-        dc = dc[:-1]
-        dc = dc[:-4] if '[0m' in dc else dc
-
-        if dc == '0' or dc == 'nan':
-            null_slices.append(n_slice)
-        else:
-            try:
-                gm_dice.append(float(dc))
-            except ValueError:
-                gm_dice.append(float(dc[:-4]))
+        try:
+            n_slice, dc = line.split(' ')
```

Comment:
You'd better avoid the try/catch and instead use a match/continue while migrating this code.

```
line = line.rstrip()
m = re.match(r"^(?P<term_color>\x1B[31;1m)?(?P<n_slice>[0-9]+) (?P<dc>[0-9.]+)(?P<term_clear>\x1B[0m)?$", line)
if m is None:
    continue
n_slice = int(m.group("n_slice"))
dc = float(m.group("dc"))
```
There, the code is much more straightforward to understand.
I can do it if you'd like.

Reply:
thanks a lot for those suggestions! agreed with all (incl. the "I can do it if you'd like" ;-) )

Reply:
I won't do that last one now as it's in the testing code and it takes more than a few minutes to figure out what it's expected to parse...